src/mappings/bank/supply.ts

```diff
@@ -0,0 +1,47 @@
+import type {
+  Coin,
+  CosmosBlock,
+} from "@subql/types-cosmos";
+import type { QueryTotalSupplyResponse } from "cosmjs-types/cosmos/bank/v1beta1/query";
+import { Supply } from "../../types";
+import { stringify } from "../utils";
+
+export const getSupplyRecord = (supply: Coin, block: CosmosBlock): Supply => Supply.create({
+  id: `${supply.denom}@${block.block.id}`,
+  denom: supply.denom,
+  amount: BigInt(supply.amount),
+  blockId: block.block.id,
+});
```

Comment:
Would you mind rewriting this as a named function instead of an anonymous function variable; it will improve readability and log stack traces.